app/src/main/java/com/github/sdpcoachme/DashboardActivity.kt

```diff
@@ -61,26 +72,60 @@ class DashboardActivity : ComponentActivity() {
             }
         }
     }
+
+    /**
+     * Create an activity for result : display window to request asked permission.
+     * If granted, launches the callback (here getDeviceLocation(...) which retrieves the user's
+     * location). The contract is a predefined "function" which takes a permission as input and
+     * outputs if the user has granted it or not.
+     */
+    private val requestPermissionLauncher =
+        registerForActivityResult(ActivityResultContracts.RequestPermission()) {
+                isGranted: Boolean -> if (isGranted) {
+                    mapViewModel.getDeviceLocation(fusedLocationProviderClient)
+                } else {
+                    // TODO Permission denied or only COARSE given
```

Comment:
We need to decide what to do if the user doesn't grant permissions for location. See #68